### PR TITLE
Front: Customer dashboard (SH-283)

### PR DESCRIPTION
**Pull request by Pikkupomo**
Add missing Customer Dashboard
Also:
* Fix default login URL, it was pointing to a wrong place
* Add Customer information Dashboard
  * Fix URL's inside Customer Information app to be more consistent
* Add Order History Dashboard
* Add Saved Carts Dashboard

All old URL's are still working.

### CODE REVIEW

File: shuup/front/templates/shuup/front/dashboard/dashboard.jinja [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+{% endblock %}
+
+{% block content %}
+<div class="container customer-dashboard">
+    <div class="row">
+        <div class="col-md-3 section-navigation">
+            <div class="sidebar-list">
+                <a class="sidebar-list-item{% if request.path == '/dashboard/' %} active{% endif %}" href="{{ url("shuup:dashboard") }}"><i class="fa fa-tachometer"></i> {% trans %}Dashboard{% endtrans %}</a>
+                {% for menu_obj in menu_items %}
+                {% set active = request.path.startswith(menu_obj.url) %}
```
Comment (tulimaki) [inline shuup/front/templates/shuup/front/dashboard/dashboard.jinja]: You should have the current item in the context, right?

File: shuup/front/templates/shuup/front/dashboard/dashboard.jinja [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        {{ macros.render_breadcrumb_item(None,  _("Dashboard"), True) }}
+    {% endcall %}
+{% endblock %}
+
+{% block content %}
+<div class="container customer-dashboard">
+    <div class="row">
+        <div class="col-md-3 section-navigation">
+            <div class="sidebar-list">
+                <a class="sidebar-list-item{% if request.path == '/dashboard/' %} active{% endif %}" href="{{ url("shuup:dashboard") }}"><i class="fa fa-tachometer"></i> {% trans %}Dashboard{% endtrans %}</a>
```
Comment (tulimaki) [inline shuup/front/templates/shuup/front/dashboard/dashboard.jinja]: Is this `request.path == '/dashboard/'` same as there is no item in context? Point beign that these comparisons to path looks sketcy.

File: shuup/front/templates/shuup/front/dashboard/dashboard.jinja [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{% extends "shuup/front/base.jinja" %}
+{% block title %}{% trans %}My Dashboard{% endtrans %}{% endblock %}
+
+{% block content_title %}{% trans %}Dashboard{% endtrans %}{% endblock %}
+
+{% block breadcrumb %}
+    {% call macros.breadcrumb() %}
+        {{ macros.render_breadcrumb_item("/", _("Home")) }}
+        {{ macros.render_breadcrumb_item(None,  _("Dashboard"), True) }}
```
Comment (tulimaki) [inline shuup/front/templates/shuup/front/dashboard/dashboard.jinja]: You could show current item at the breadcrumb.

File: shuup/front/utils/dashboard.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        template = get_template(self.template)
+        return mark_safe(template.render(context=self.get_context()))
+
+    @property
+    def css_class(self):
+        return "col-md-%d" % self.size.value
+
+    def show_on_menu(self):
+        """
+        Defines if the item is shown in the main dashboard menu
```
Comment (tulimaki) [inline shuup/front/utils/dashboard.py]: Also the rest of the properties should have docstrings. Also this one seems to be missing something from return.

File: doc/provides.rst [hunk 5]
```diff
@@ -97,6 +97,10 @@ Core
     Admin module classes. Practically all of the functionality in the admin is built
     via admin modules.
 
+``customer_dashboard_items``
+    Classes to parse customer dashboard links from. These are subclasses of
```
Comment (tulimaki) [inline doc/provides.rst]: Maybe not links?

File: shuup_tests/front/test_customer_information.py [hunk 6]
```diff
@@ -7,28 +7,21 @@
...
-from shuup.front.apps.customer_information.views import (
-    change_password, CompanyEditView, CustomerEditView
-)
 from shuup.testing.factories import get_default_shop
 from shuup.testing.soup_utils import extract_form_fields
-from shuup.testing.utils import apply_request_middleware
 from shuup_tests.utils import SmartClient
 from shuup_tests.utils.fixtures import (
-    regular_user, REGULAR_USER_PASSWORD, REGULAR_USER_USERNAME
+    REGULAR_USER_PASSWORD, REGULAR_USER_USERNAME
```
Comment (tulimaki) [inline shuup_tests/front/test_customer_information.py]: maybe removing `regular_user` from here was a mistake.

File: shuup/front/apps/personal_order_history/urls.py [hunk 7]
```diff
@@ -11,8 +11,8 @@
 
 urlpatterns = patterns(
     '',
-    url(r'^orders/$', views.OrderListView.as_view(),
+    url(r'^order-history/$', views.OrderListView.as_view(),
```
Comment (tulimaki) [inline shuup/front/apps/personal_order_history/urls.py]: This change is probably breaking some hard coded notification scrips in some instances. Should be included in changelog.

File: shuup/front/static_src/less/front/footer.less [hunk 8]
```diff
@@ -7,7 +7,7 @@ footer {
         background: @footer-bg;
 
         .container {
-            padding: 25px 0;
+           // padding: 25px 0;
```
Comment (tulimaki) [inline shuup/front/static_src/less/front/footer.less]: Maybe remove the whole definition. No need to preserve this code.

File: shuup/regions/resources.py [hunk 9]
```diff
@@ -17,80 +17,94 @@
...
+        $regionCodeField.closest("div.form-group").hide();
     }
+}
 
-    function updateBillingRegion() {
-        var country = $("%(billing_field_prefix)s-country").val();
-        var regionsData = window.REGIONS[country];
-        handleRegionFields("%(billing_field_prefix)s-region", "%(billing_field_prefix)s-region_code", regionsData);
+function updateRegion(field) {
+    var country_field = field + "-country";
```
Comment (tulimaki) [inline shuup/regions/resources.py]: CamelCase for JS variables.

**Comment by tulimaki** [comment]
First commit message is bad. The base dashboard functionality seems to be missing tests. Third commit message is bad. I guess there is something else happening too in that third commit. Also the addressbook view is missing tests and probably could be in it's own commit. While fixing these two commit message check the other ones also. Also all the changelogs is missing.

File: shuup/front/apps/customer_information/templates/shuup/customer_information/addressbook/index.jinja [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+{% block dashboard_content %}
+<div class="user-account-address">
+    <h1>{% trans %}Default Addresses{% endtrans %}
+    <a class="btn btn-primary pull-right" role="button" href="{{ url("shuup:customer_edit") }}">{% trans %}Edit{% endtrans %}</a>
+    </h1>
+    <div class="col-md-6">
+        {% call content_block(_("Default Billing Address")) %}
+            {% set addr = customer.default_billing_address %}
+            {{ render_address(addr) }}
```
Comment (tulimaki) [inline shuup/front/apps/customer_information/templates/shuup/customer_information/addressbook/index.jinja]: Can you just pass the customer.default_billing_address without setting any variables? Also the `render_address` fails little in case the address is not set.

**Comment by tulimaki** [comment]
Dashboard probably could be available from top menu.

**Comment by tulimaki** [comment]
Some of the new stuff like customer information-app is probably missing some translations.

File: shuup/front/apps/customer_information/views.py [hunk 11]
```diff
@@ -5,34 +5,45 @@
...
 from shuup.utils.form_group import FormGroup
 from shuup.utils.importing import cached_load
 
-from .forms import CompanyContactForm, PersonContactForm
+from .forms import CompanyContactForm, PersonContactForm, SavedAddressForm
 from .notify_events import CompanyAccountCreated
 
 
-def change_password(request):
+class PasswordChangeView(DashboardViewMixin, TemplateView):
```
Comment (tulimaki) [inline shuup/front/apps/customer_information/views.py]: Could you tell something about this change in commit message. PRobably could be a separate commit?

File: shuup/front/templates/shuup/front/dashboard/dashboard.jinja (hunk 3 above)
Comment (Pikkupomo) [inline shuup/front/templates/shuup/front/dashboard/dashboard.jinja]: there is no current item in this view as this is kind of `index` view

File: shuup/front/apps/personal_order_history/urls.py (hunk 7 above)
Comment (Pikkupomo) [inline shuup/front/apps/personal_order_history/urls.py]: okay!

File: shuup/front/apps/customer_information/templates/shuup/customer_information/customer_dashboard_item.jinja [hunk 12]
```diff
@@ -0,0 +1,49 @@
+{% extends "shuup/front/dashboard/dashboard_base.jinja" %}
+{%- from "shuup/front/macros/dashboard.jinja" import content_block -%}
+
+{% macro render_address(addr) %}
```
Comment (tulimaki) [inline shuup/front/apps/customer_information/templates/shuup/customer_information/customer_dashboard_item.jinja]: Is this same macro as in addresbook index on purpose?

File: shuup/front/apps/customer_information/templates/shuup/customer_information/customer_dashboard_item.jinja [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+            <p><b>{% trans %}Birth Date{% endtrans %}:</b> {{ customer.birth_date or "Not specified" }}</p>
+            <p><b>{% trans %}Gender{% endtrans %}:</b> {{ customer.gender or "Not specified" }}</p>
+        {% endif %}
+        <p><b>{% trans %}Marketing Permission{% endtrans %}:</b> {{ _("Yes") if customer.marketing_permission else _("No") }}</p>
+    {% endcall %}
+    </div>
+    <div class="col-md-4">
+    {% call content_block(_("Default Billing Address")) %}
+        {% set addr = customer.default_billing_address %}
+        {{ render_address(addr) }}
```
Comment (tulimaki) [inline shuup/front/apps/customer_information/templates/shuup/customer_information/customer_dashboard_item.jinja]: This might not work if customer does not have address
